solr/core/src/java/org/apache/solr/core/FileSystemConfigSetService.java

```diff
@@ -287,9 +287,11 @@ public Long getCurrentSchemaModificationVersion(
   }
 
   protected Path getConfigDir(String configName) throws IOException {
-    String configSetDirPath = configSetBase.toFile().getCanonicalPath() + File.separator;
-    String path = Paths.get(configSetBase.toString(), configName).toFile().getCanonicalPath();
-    if (!path.startsWith(configSetDirPath)) {
+    String path =
+        Paths.get(configSetBase.toString(), FilenameUtils.getName(configName))
```

Comment:
If configName were "foo/bar", there shouldn't be anything fundamentally wrong with that.